Documentation/api-reference.md

```diff
@@ -0,0 +1,529 @@
+# Adobe Journey Optimizer - Decisioning API reference
+
+## Prerequisites
+
+Refer to the [Getting Started Guide](getting-started.md).
+
+## API reference
+
+- [clearPropositions](#clearPropositions)
+- [extensionVersion](#extensionVersion)
+- [getPropositions](#getPropositions)
+- [onPropositionsUpdate](#onPropositionsUpdate)
+- [registerExtension](#registerExtension)
+- [resetIdentities](#resetIdentities)
+- [updatePropositions](#updatePropositions)
+
+## Public classes
+
+- [DecisionScope](#DecisionScope)
+- [Proposition](#Proposition)
+- [Offer](#Offer)
+- [OfferType](#OfferType)
+
+## API reference
+
+## clearPropositions
+
+This API clears out the client-side in-memory propositions cache.
+
+### Java
+
+#### Syntax
+
+```java
+public static void clearCachedPropositions()
+```
+
+#### Example
+
+```java
+Optimize.clearCachedPropositions();
+```
```

Comment:
Are we not adding Kotlin syntax/ examples?

Reply:
Other extensions have only added Kotlin syntax for initialization and registration. Following the same pattern here. Also the Java and Kotlin class are the same

Reply:
As discusses, let's add the Kotlin syntax/ examples for the APIs.